qemu/target-arm/translate.c

```diff
@@ -10407,7 +10407,20 @@ static void disas_thumb_insn(CPUARMState *env, DisasContext *s) // qq
 
     // Unicorn: trace this instruction on request
     if (HOOK_EXISTS_BOUNDED(s->uc, UC_HOOK_CODE, s->pc)) {
-        gen_uc_tracecode(tcg_ctx, 2, UC_HOOK_CODE_IDX, s->uc, s->pc);
+        // determine instruction size (Thumb or Thumb2)
+        insn = arm_lduw_code(env, s->pc, s->bswap_code);
```

Comment:
can you optimize this code, so we can avoid this calculation again on the line 10428?